server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchSourcePhase.java

```diff
@@ -60,8 +60,9 @@ private void hitExecute(FetchSourceContext fetchSourceContext, HitContext hitCon
                 final boolean nestedHit = hitContext.hit().getNestedIdentity() != null;
                 Source source = hitContext.source();
 
-                // If this is a parent document and there are no source filters, then add the source as-is.
-                if (nestedHit == false && fetchSourceContext.hasFilter() == false) {
+                // If this is a parent document and there are no source filters and include_vector is set as true, then add the source
+                // as-is.
+                if (nestedHit == false && sourceFilter.isEmpty()) {
```

Comment:
This is a nit, but this comment is a bit misleading. `sourceFilter` can be empty regardless of the value of `includeVectors`, for example when there are no vector fields to filter out. 

At this point in the execution, we don't care _why_ `sourceFilter` is empty, just handling when it is. The comment should reflect that.